### PR TITLE
Implementation NNCF support for face recognition models

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
Added configs for MobileNet and Landnet sparsification and quantization experiments (more about NNCF in https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/nncf).
The compressed models results are shown in the table 
Screenshot from 2019-10-29 17-01-03

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
Please rebase your develop branch to original develop
https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa

File: pytorch_toolkit/face_recognition/README.md [hunk 1]
```diff
@@ -118,3 +119,89 @@ uncleaned version of the [MegaFace](http://megaface.cs.washington.edu/participat
 ## Face Recognition Demo
 
 1. For setting up demo, please go to [Face Recognition demo with OpenVINO Toolkit](./demo/README.md)
+
+## Train compressed model using [NNCF](https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/nncf)
+
+
+[NNCF](https://github.com/opencv/openvino_training_extensions/tree/develop/pytorch_toolkit/nncf) - is a framework for neural network compression using quantization and sparsification algorithms.
+
+To use NNCF as a standalone package refer to this [manual](https://github.com/opencv/openvino_training_extensions/blob/develop/pytorch_toolkit/nncf/docs/PackageUsage.md).
```
Comment (AlexanderDokuchaev) [inline pytorch_toolkit/face_recognition/README.md]: Installation of nncf should be done in `init_venv.sh`, because you will get `Import error` when try to train models without `nncf`. 
Try to use `pip3 install -e ../nncf`, may be it will work.

File: pytorch_toolkit/face_recognition/train.py [hunk 2]
```diff
@@ -217,11 +263,16 @@ def main():
     parser.add_argument('--snap_to_resume', type=str, default=None, help='Snapshot to resume.')
     parser.add_argument('--weighted', action='store_true')
 
+    # compression parameters
```
Comment (AlexanderDokuchaev) [inline pytorch_toolkit/face_recognition/train.py]: Dont split arguments by comment

File: pytorch_toolkit/face_recognition/train.py [hunk 3]
```diff
@@ -90,7 +95,16 @@ def train(args):
 
     model = models_backbones[args.model](embedding_size=args.embed_size,
                                          num_classes=dataset.get_num_classes(), feature=False)
+
+    set_dropout_fn = model.set_dropout_ratio
```
[the remark this answers is not in the document]
Comment (GalyaZalesskaya) [reply pytorch_toolkit/face_recognition/train.py]: It is needed for training compressed models without dropout

[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply pytorch_toolkit/face_recognition/train.py]: I mean that why dont use 
`model.set_dropout_ratio(0.1)`
insdead of 
```
set_dropout_fn = model.set_dropout_ratio
...
set_dropout_fn(0.1)
```

[the remark this answers is not in the document]
Comment (GalyaZalesskaya) [reply pytorch_toolkit/face_recognition/train.py]: Because if we compress model, we wrap it with compession_algo
` model = compression_algo.model (106 line)` 
and should call model.model.set_dropout_fn(0.1). But if we train simple model, we should call model.set_dropout_fn(0.1).
This hint is used only to write universal code for several conditions.

[the remark this answers is not in the document]
Comment (AlexanderDokuchaev) [reply pytorch_toolkit/face_recognition/train.py]: Hm, ok.

**Comment by AlexanderDokuchaev** [comment]
@GalyaZalesskaya Which version of pytorch did you used? 
Face_recognition have required [0.4.1](https://github.com/opencv/openvino_training_extensions/blob/develop/pytorch_toolkit/face_recognition/requirements.txt#L10), but nncf 1.0 or higher.

**Comment by AlexanderDokuchaev** [comment]
@GalyaZalesskaya 
You forget add config files with new names

**Comment by GalyaZalesskaya** [comment]
I used torch 1.2 (for CUDA 10.0), but I noticed now that after installation of FR and NNCF requirements there is still torch 0.4.1. Should I add the line into init_venv.sh by force
`pip install torch==1.2`?
But CUDA and Torch versions should be compatible.

Sorry for delaying. I found some issue which appeared after rebasing and I'm still trying to solve it.

**Comment by AlexanderDokuchaev** [comment]
@GalyaZalesskaya 
> Sorry for delaying. After you comment I found an issue which appeared after rebasing and I'm still trying to solve it.

No problem

> I used torch 1.2 (for CUDA 10.0), but I noticed now that after installation of FR and NNCF requirements there is still torch 0.4.1. Should I add the line into init_venv.sh by force
pip install torch==1.2?

No, I'll try to check it by myself and merge pull request on this week.
